script.js

```diff
@@ -30,6 +32,39 @@ function setStyles(element) {
 	element.style.left = randInRange(0,leftMax) + 'px';
 	element.style.top = randInRange(0,topMax) + 'px';
 	element.style.backgroundColor = 'rgba(' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + Math.random() + ')';
+
+	// using setTimeout, run setStyles on the box once every second, continuously
+	/*var repeater = setTimeout(function(){
+		setStyles(element);
+	},1000);*/
+}
+
+// create a new loop that smoothly animates the position of the box, moving it 60 times per second
+function moveBox(element) {
+    var leftPos = 0,
+    	topPos = 0,
+		divWidth = element.offsetWidth,
+		divHeight = element.offsetHeight,
+		leftMax = windowWidth - divWidth,
+		topMax = windowHeight - divHeight,
```

Comment:
Width and height of the div and the max left and top positions aren't going to change every 60th of a second. It's much more efficient to pull these out and calculate once.